backend/infrahub/core/schema/schema_branch.py

```diff
@@ -1571,7 +1577,10 @@ def add_hierarchy_node(self) -> None:
                 if "parent" not in node.relationship_names:
                     node.relationships.append(
                         self._get_hierarchy_parent_rel(
-                            peer=node.parent, hierarchical=node.hierarchy, read_only=read_only
+                            peer=node.parent,
+                            hierarchical=node.hierarchy,
+                            read_only=read_only,
+                            optional=node.parent in [node_name] + self.generic_names,
```

Comment:
This looks like a bug to me. The optional parameter is supposed to be a `str` but we are sending a `bool + list[str]` which would give a `TypeError`. It's something that mypy would have caught if this file wasn't ignored.

However it looks like the line is covered by tests (https://coveralls.io/builds/71927799/source?filename=backend%2Finfrahub%2Fcore%2Fschema%2Fschema_branch.py#L1583), so I don't quite understand how this is working?

I'd wager that the intention is that optional should be a boolean?